msmbuilder/featurizer/featurizer.py

```diff
@@ -307,6 +307,77 @@ def partial_transform(self, traj):
         return d ** self.exponent
 
 
+class FunctionFeaturizer(Featurizer):
+    """Featurizer based on arbitrary functions.
+
+    This featurizer transforms a dataset containing MD trajectories into
+    a vector dataset by representing each frame in each of the MD trajectories
+    by a vector the output of the function.
+
+    Parameters
+    ----------
+    function : function
+        Instantiation of the function. The function should accept
+        a mdtraj.Trajectory object as the first argument.
+    *args : list
+        arguments to pass to the function MINUS the trajectory keyword
+    **kwargs: kwargs
+        key word arguments to pass to the function MINUS the trajectory
+        keyword
+
+    Notes
+    ----------
+    This Featurizer assumes that the function takes in the trajectory object
+    as the first argument.
+
+    Examples
+    --------
+    >>> function = compute_dihedrals
+    >>> f = FunctionFeaturizer(function, indices=[[0,1,2,3]])
+    >>> results = f.transform(dataset)
+    """
+
+    def __init__(self, function, *args, **kwargs):
+        if hasattr(function, '__call__'):
```

Comment:
I think using the builtin callable covers more use-cases here

https://docs.python.org/3.4/library/functions.html#callable